backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/PatientRegistrationController.java

```diff
@@ -0,0 +1,215 @@
+package ca.bc.gov.hlth.hnweb.controller;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import ca.bc.gov.hlth.hnweb.converter.hl7v3.GetDemographicsConverter;
+import ca.bc.gov.hlth.hnweb.model.rest.StatusEnum;
+import ca.bc.gov.hlth.hnweb.model.rest.enrollment.GetPersonDetailsResponse;
+import ca.bc.gov.hlth.hnweb.model.rest.patientregistration.PatientRegisterModel;
+import ca.bc.gov.hlth.hnweb.model.rest.patientregistration.PatientRegistrationRequest;
+import ca.bc.gov.hlth.hnweb.model.rest.patientregistration.PatientRegistrationResponse;
+import ca.bc.gov.hlth.hnweb.model.v3.GetDemographicsRequest;
+import ca.bc.gov.hlth.hnweb.model.v3.GetDemographicsResponse;
+import ca.bc.gov.hlth.hnweb.persistence.entity.AffectedPartyDirection;
+import ca.bc.gov.hlth.hnweb.persistence.entity.IdentifierType;
+import ca.bc.gov.hlth.hnweb.persistence.entity.Transaction;
+import ca.bc.gov.hlth.hnweb.persistence.entity.pbf.PatientRegister;
+import ca.bc.gov.hlth.hnweb.security.TransactionType;
+import ca.bc.gov.hlth.hnweb.service.EnrollmentService;
+import ca.bc.gov.hlth.hnweb.service.PatientRegistrationService;
+
+/**
+ * Handles request related to R70 Patient Registration.
+ */
+@RequestMapping("/patient-registration")
+@RestController
+public class PatientRegistrationController extends BaseController {
+
+	private static final Logger logger = LoggerFactory.getLogger(PatientRegistrationController.class);
+
+	@Autowired
+	private EnrollmentService enrollmentService;
+
+	@Autowired
+	private PatientRegistrationService patientRegistrationService;
+
+	@PostMapping("/get-patient-registration")
+	public ResponseEntity<PatientRegistrationResponse> getPatientRegistration(
+			@Valid @RequestBody PatientRegistrationRequest patientRegistrationRequest,
+			HttpServletRequest request) {
+
+		logger.info("Patient Registration request: {} ", patientRegistrationRequest.getPhn());
```

Comment:
Do we need to validate that the provided Payee number is actually the user's assigned payee number? Otherwise they can access other PHNs by just sending a different Payee number (e.g. through a direct API call).

Reply:
Can cover as a separate story once https://dev.azure.com/cgi-vic-hlth/HNWeb/_workitems/edit/7102  is done.